### PR TITLE
Support variety of data inputs for wind barbs

### PR DESCRIPTION
**Pull request by ahuang11**
Based on feedback from
https://github.com/holoviz/holoviews/pull/5797

### CODE REVIEW

File: geoviews/element/geo.py [hunk 1]
```diff
@@ -353,29 +353,30 @@ class WindBarbs(_Element, Selection2DExpr, HvGeometry):
...
-            vs = data[vdims[1]]
+        if kdims is None:
+            kdims = ['x', 'y']
+        if vdims is None:
+            vdims = ['u', 'v']
+        dataset = Dataset(data, kdims=kdims, vdims=vdims, **params)
+        us, vs = (dataset.dimension_values(i) for i in range(2, 4))
 
         uv_magnitudes = np.hypot(us, vs)  # unscaled
         radians = np.pi / 2 - np.arctan2(-vs, -us)
```
Comment (Hoxbro) [inline geoviews/element/geo.py:364]: Let's add a comment that this is meteorological coordinates.

File: geoviews/tests/plotting/mpl/test_chart.py [hunk 2]
```diff
@@ -54,7 +54,7 @@ def test_windbarbs_dataset(self):
         np.testing.assert_almost_equal(mpl_barbs.v.data, V.T.flatten())
 
     def test_windbarbs_from_uv(self):
-        x = np.linspace(-1, 1, 4)
+        x = np.linspace(-1, 1, 2)
```
Comment (Hoxbro) [inline geoviews/tests/plotting/mpl/test_chart.py:57]:
```suggestion
        x = np.array([-1, 1])
```

Nit again.

Is there a reason for the change in the input array?

**Review by Hoxbro — COMMENTED** [review]
Do you think we should add an extra test where `vdims` is `[hv.Dimension("u"), hv.Dimension("v")]`.

I don't know if this will give any problems but using `hv.Dataset` will flatten the array. I can see you already have flattened the result array because of this. 

``` python
import numpy as np
import holoviews as hv

x = np.array([-1, 1])
X, Y = np.meshgrid(x, x)
U, V = 10 * X, 2 * Y
U

ds = hv.Dataset((X, Y, U, V), kdims=["x", "y"], vdims=["u", "v"])
ds.dimension_values("u")
```
image

File: geoviews/tests/plotting/mpl/test_chart.py (hunk 2 above)
Comment (ahuang11) [reply geoviews/tests/plotting/mpl/test_chart.py]: I was testing why it was failing so I reduced the size so I can compare & contrast more easily; I changed it back.

**Review by Hoxbro — APPROVED** [review]
Let us keep this PR open to sync if more changes are added in https://github.com/holoviz/holoviews/pull/5797.

**Pull request merged**